app/src/main/java/com/firebase/uidemo/database/ChatActivity.java

```diff
@@ -131,26 +136,50 @@ public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
     }
 
     private void attachRecyclerViewAdapter() {
-        Query lastFifty = mChatRef.limitToLast(50);
-        mAdapter = new FirebaseRecyclerAdapter<Chat, ChatHolder>(
-                Chat.class, R.layout.message, ChatHolder.class, lastFifty) {
+        mAdapter = new FirebaseIndexRecyclerAdapter<Chat, ChatHolder>(
```

Comment:
I have a really strong preference to keeping this since I was able to find issues with the index stuff by testing with this branch.